### ISSUE TITLE
你好，请问你的cincgan复现成功了吗

### ISSUE THREAD

**Issue by Tomlk** [jianghongjie328/cincgan#1]
你好，请问你的cincgan复现成功了吗，我发现github上的cincgan第二轮大多都有问题，想和你探讨下细节。

**Comment by zookbarry** [comment]
你好，不好意思，我的cincgan在上次上传代码之后由于学业压力就一直没有继续，所以outter部分一直没有进行复现，不过一个月后打算继续进行复现，到时很希望可以和您讨论学习。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2021年11月28日(周日) 晚上9:21
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;;
主题: [jianghongjie328/cincgan] 你好，请问你的cincgan复现成功了吗 (Issue #1)




 
你好，请问你的cincgan复现成功了吗，我发现github上的cincgan第二轮大多都有问题，想和你探讨下细节。
 
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Tomlk** [comment]
你好，我也要向你学习，我对原论文的outter细节一直有疑问，原文
”We then jointly fine-tune the LR to HR model (shown as
LR!HR in Fig. 2). We initialize our SR network by pub-
licly available EDSR model3. We set parameters in (10) as
�1 = 10,�2 = 5 and �3 = 2. The optimizer is set almost
the same as training the LR!clean LR model, except for
we initialize learning rate with 10�-4. As to the weight of
identity loss LLR in (5), we set w2 = 1. At each iteration, idt
we update (5) and (10) in turn. We first train G1 and G2 to update the LR!clean LR network. We then train G1 , S R and G3 simultaneously to update the LR!HR network.“。这里提到在outter将w2改为1，也就是说inner会继续训练，但是把初始化学习率改为了10^-4，这除了outter层学习率从10^-4开始，那么inner层还是2 × 10^{-4}开始嘛，还是说从当前已衰减过的学习率开始(2 × 10^4每40000iter衰减一半，最终400000iter后衰减的那个数值)。

**Comment by zookbarry** [comment]
你好，不好意思，我对于这方面经验比较欠缺，不过我参考了一下https://github.com/Junshk/CinCGAN-pytorch的实现，其中对于您这个问题它的实现时使用了1*10^-4做两个部分的优化。个人看法是它做端到端训练为的是探寻到一个组优秀的参数来使inner和outter两个部分协同进行超分辨率放大时，这组参数是较优的，所以我想虽然在inner层中已经在一个很低的损失率下获得了收敛，但是在inner和outter一同做端到端训练时，采用一个较大的初始损失率也许有利于探索更大的参数空间，更有利于得到这组较优参数。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jianghongjie328/cincgan"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月29日(星期一) 上午10:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jianghongjie328/cincgan] 你好，请问你的cincgan复现成功了吗 (Issue #1)





 
你好，我也要向你学习，我对原论文的outter细节一直有疑问，原文
 ”We then jointly fine-tune the LR to HR model (shown as
 LR!HR in Fig. 2). We initialize our SR network by pub-
 licly available EDSR model3. We set parameters in (10) as
 �1 = 10,�2 = 5 and �3 = 2. The optimizer is set almost
 the same as training the LR!clean LR model, except for
 we initialize learning rate with 10�-4. As to the weight of
 identity loss LLR in (5), we set w2 = 1. At each iteration, idt
 we update (5) and (10) in turn. We first train G1 and G2 to update the LR!clean LR network. We then train G1 , S R and G3 simultaneously to update the LR!HR network.“。这里提到在outter将w2改为1，也就是说inner会继续训练，但是把初始化学习率改为了10^-4，这除了outter层学习率从10^-4开始，那么inner层还是210^-4开始嘛，还是说从当前已衰减过的学习率开始(210^4每40000iter衰减一半，最终400000iter后衰减的那个数值)。
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Tomlk** [comment]
谢谢，我训练完inner后，发现最开始能够去模糊和去噪，但会有颜色失真，但是随着稳定，生成的clean lr会再次变得模糊，这时候没有颜色失真，虽然这些噪声相对于原始输入有了少许改变。根据基于CinCGAN改进的mCinCGAN论文”Multiple Cycle-in-Cycle Generative Adversarial Networks for Unsupervised Image Super-Resolution“ 所述：“They claim that the identity loss can help preserving the color of input images. However, this loss will also compel G1(x) = x, which is harmful for G1 to restore the clean version of the LR input。”即inner层的idt损失由于w2初始为5较大，为了维持颜色稳定，也会破坏G1的去模糊效果，不知道你稳定后训练出的clean LR是否这样？

基于此，CinCGAN作者在训练outer时把w2从5改为1，联合训练G1和微调EDSR模型。

你好，我最近在致力于复现CInCGAN并做出改进，能否加微信深入交流？我的微信：lk475207136。

**Comment by zookbarry** [comment]
是的，我也遇到了这样的问题，我训练的感觉是一开始先生成的图像主要是图像轮廓，到损失接近0.1时，图像基本通过G1可以还原，但是，但是接下来从0.1-0.04图像的噪声减少程度基本就没有了什么变化，我猜想会不会时无噪声图像的分布没有捕获好，我试着去增加了gan的discriminator的训练次数，从1次D，1次G，改为1次G，两次D，但是效果并不明显。本来还在猜想会不会跟着outter训练一下问题就解决了，但想了一下会不会太玄学了，没先到还真是。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jianghongjie328/cincgan"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月29日(星期一) 晚上6:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jianghongjie328/cincgan] 你好，请问你的cincgan复现成功了吗 (Issue #1)





 
谢谢，我训练完inner后，发现最开始能够去模糊和去噪，但会有颜色失真，但是随着稳定，生成的clean lr会再次变得模糊，这时候没有颜色失真，虽然这些噪声相对于原始输入有了少许改变。根据基于CinCGAN改进的mCinCGAN论文”Multiple Cycle-in-Cycle Generative Adversarial Networks for Unsupervised Image Super-Resolution“ 所述：“They claim that the identity loss can help preserving the color of input images. However, this loss will also compel G1(x) = x, which is harmful for G1 to restore the clean version of the LR input。”即inner层的idt损失由于w2初始为5较大，为了维持颜色稳定，也会破坏G1的去模糊效果，不知道你稳定后训练出的clean LR是否这样？
 
基于此，CinCGAN作者在训练outer时把w2改为1，联合训练G1和微调EDSR模型。
 
你好，我最近在致力于复现CInCGAN并做出改进，能否加微信深入交流？我的微信：lk475207136。
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
